torch/_inductor/fx_passes/quantization.py

```diff
@@ -142,3 +164,196 @@ def qconv(match: Match, *args, **kwargs):
 
 def register_quantization_lowerings():
     _register_quantized_conv_lowering(quantize_conv_output_pattern)
+
+
+def _is_valid_dequant_conv2d_pattern(match):
+    conv_node = match.output_node()
+    assert conv_node.target is aten.convolution.default
+    input_meta_value = conv_node.args[0].meta.get("val")
+    weight_meta_value = conv_node.args[1].meta.get("val")
+    for meta_value in [input_meta_value, weight_meta_value]:
+        if (
+            meta_value is None
+            or meta_value.device.type != "cpu"
+            or meta_value.dim() != 4
+        ):
+            # Only support conv2d now
+            return False
+
+    (
+        to_fp32_node,
+        sub_node,
+        mul_node,
+    ) = match.nodes[0:3]
+
+    assert to_fp32_node.target is torch.ops.prims.convert_element_type.default
+    assert sub_node.target is torch.ops.aten.sub.Tensor
+    assert mul_node.target is torch.ops.aten.mul.Tensor
+    if (
+        len(list(to_fp32_node.users)) != 1
+        or len(list(sub_node.users)) != 1
+        or len(list(mul_node.users)) != 1
+    ):
+        # Ensure the dequant pattern only has 1 user
+        # since we will delete the dequant pattern here
+        return False
+    return True
+
+
+def _register_qconv_weight_prepack_pass(pattern):
+    @register_freezing_graph_pattern(
+        pattern,
+        extra_check=_is_valid_dequant_conv2d_pattern,
+        pass_number=1,  # pass_number=1, ensure it's behand dequant promotion pass
+    )
+    def qconv_weight_prepack(match: Match, *args, **kwargs):
+        """
+        Macth the pattern:
+        int8 activation
+          |
+        dequant_per_tensor
+          |
+        Conv2d <- optional(aten.clone.default) <- dequant_per_channel
+
+        Insert weight prepack node and change the pattern to:
+        int8 activation
+          |
+        onednn.qconv2d_pointwise <- onednn.qconv_prepack <- optional(aten.clone.default) <- dequant_per_channel
+        """
+        has_clone_to_channel_last_node_in_pattern = any(
+            node.target == aten.clone.default for node in match.nodes
+        )
+
+        (
+            to_fp32_node,
+            sub_node,
+            mul_node,
+            dequant_per_channel,
+        ) = match.nodes[0:4]
```

Comment:
I think if we have a string to node mapping then we can retrieve the node by string name, like what we have in TorchScript, and it will be more robust than this as well

Reply:
Yean, we shouldn't write it as this. And I have changed it into
```
conv_node = match.output_node()
mul_node = conv_node.args[0]
sub_node = mul_node.args[0]
to_fp32_node = sub_node.args[0]
```
Since the pattern only has 1 output node as `conv_node`, so these node order are guaranteed. 
Alternately, there is another way to use `filter_nodes` method with node target as 
```
conv_nodes = filter_nodes(match.nodes, aten.convolution.default)
mul_nodes = filter_nodes(match.nodes, aten.mul.Tensor)
sub_nodes = filter_nodes(match.nodes, aten.sub.Tensor)
```
It will return all the nodes with the target op in the matched pattern.